src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -1090,6 +1090,70 @@ public async Task EchoServerStreaming_DifferentMessageSize_Success(int messageSi
             connection.Dispose();
         }
 
+        [Fact]
+        [OuterLoop("Uses Task.Delay")]
+        public async Task RequestContentStreaming_Timeout_BothClientAndServerReceiveCancellation()
+        {
+            if (UseQuicImplementationProvider == QuicImplementationProviders.Mock)
+            {
+                return;
+            }
+
+            var message = new byte[1024];
+            var random = new Random(0);
+            random.NextBytes(message);
+
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer();
+            Http3LoopbackConnection connection = null;
+            Http3LoopbackStream serverStream = null;
+
+            Task serverTask = Task.Run(async () =>
+            {
+                connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync();
+                serverStream = await connection.AcceptRequestStreamAsync();
+
+                await serverStream.HandleRequestAsync().WaitAsync(TimeSpan.FromSeconds(60));
+            });
+
+            StreamingHttpContent requestContent = new StreamingHttpContent();
+
+            using HttpClient client = CreateHttpClient();
+            client.Timeout = TimeSpan.FromSeconds(0.3); // set small timeout
+            using HttpRequestMessage request = new()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = server.Address,
+                Version = HttpVersion30,
+                VersionPolicy = HttpVersionPolicy.RequestVersionExact,
+                Content = requestContent
+            };
+
+            var responseTask = client.SendAsync(request).WaitAsync(TimeSpan.FromSeconds(60));
+
+            Stream requestStream = await requestContent.GetStreamAsync();
+            // Send headers
+            await requestStream.FlushAsync();
+
+            await requestStream.WriteAsync(message).AsTask().WaitAsync(TimeSpan.FromSeconds(10));
```

Comment:
How do we make sure that the 0.3 seconds timeout of `HttpClient` doesn't fire before this `WriteAsync`?